x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/documents.tsx

```diff
@@ -100,22 +88,51 @@ export const SearchIndexDocuments: React.FC = () => {
     setPagination(DEFAULT_PAGINATION);
     getMappings({ indexName: indexToShow });
   }, [indexToShow]);
-
   return (
-    <EuiPanel hasBorder={false} hasShadow={false} paddingSize="none">
-      <EuiSpacer />
-      <EuiFlexGroup direction="column">
-        <EuiFlexItem>
-          <EuiFlexGroup direction="row" alignItems="center">
-            <EuiFlexItem className="enterpriseSearchDocumentsHeader" grow={false}>
-              <EuiTitle>
-                <h2>
-                  {i18n.translate('xpack.enterpriseSearch.content.searchIndex.documents.title', {
-                    defaultMessage: 'Browse documents',
+    <>
+      <DocumentsOverview
+        dataTelemetryIdPrefix={`entSearchContent-${ingestionMethod}`}
+        searchQueryCallback={searchQueryCallback}
+        documentComponent={
+          <>
+            {isAccessControlIndexNotFound && (
+              <EuiCallOut
+                size="m"
+                title={i18n.translate(
+                  'xpack.enterpriseSearch.content.searchIndex.documents.noIndex.title',
+                  { defaultMessage: 'Access Control Index not found' }
+                )}
+                iconType="iInCircle"
+              >
+                <p>
+                  {i18n.translate('xpack.enterpriseSearch.content.searchIndex.documents.noIndex', {
+                    defaultMessage:
+                      "An Access Control Index won't be created until you enable document-level security and run your first access control sync.",
                   })}
-                </h2>
-              </EuiTitle>
-            </EuiFlexItem>
+                </p>
+              </EuiCallOut>
+            )}
+            {!isAccessControlIndexNotFound &&
+              docs.length === 0 &&
+              i18n.translate('xpack.enterpriseSearch.content.searchIndex.documents.noMappings', {
+                defaultMessage: 'No documents found for index',
+              })}
+            {!isAccessControlIndexNotFound && docs.length > 0 && (
+              <DocumentList
+                dataTelemetryIdPrefix={`entSearchContent-${ingestionMethod}`}
+                docs={docs}
+                docsPerPage={pagination.pageSize ?? 10}
+                isLoading={status !== Status.SUCCESS && mappingStatus !== Status.SUCCESS}
+                mappings={mappingData?.mappings?.properties ?? {}}
+                meta={data?.meta ?? DEFAULT_PAGINATION}
+                onPaginate={(pageIndex) => setPagination({ ...pagination, pageIndex })}
+                setDocsPerPage={(pageSize) => setPagination({ ...pagination, pageSize })}
+              />
+            )}
+          </>
+        }
+        accessControlSwitch={
+          <>
             {shouldShowAccessControlSwitcher && (
               <EuiFlexItem grow={false}>
```

Comment:
I'm not positive on this, so I would like @sphilipse opinion too.

I think we should make this prop optional and handle the `<EuiFlexItem>` in the `<DocumentsOverview>` component. 

which would make this:

```typescript
 accessControlSwitch={shouldShowAccessControlSwitcher ? (
  <AccessControlIndexSelector
    onChange={setSelectedIndexType}
    valueOfSelected={selectedIndexType}
  />) : undefined} 
```

And in the `DocumentsOverview` it would be something like:

```typescript
{accessControlSwitch && (<EuiFlexItem grow={false}>{accessControlSwitch}</EuiFlexItem>)}
```

This would hide the flex group rendering from the consumer of the `DocumentsOverview` component and make it a bit less error prone IMO.

Reply:
I would agree